Code/Framework/AzCore/Tests/Platform/Android/Tests/UtilsTests_Android.cpp

```diff
@@ -69,9 +69,14 @@ namespace UnitTest
         // Note that ConvertToAbsolutePath will perform a realpath on the result. The result of AZ::Utils::GetExecutableDirectory
         // uses AZ::Android::AndroidEnv::Get()->GetAppPrivateStoragePath() which will retrieve the storage path, but that path could
         // be symlinked, so we need to perform a real path on it before comparison
-        char realExecutableDirectory[AZ::IO::MaxPathLength];
-        ASSERT_TRUE(realpath(executableDirectory, realExecutableDirectory));
-
+        char* realExecutableDirectory = realpath(executableDirectory, nullptr);
+        ASSERT_TRUE( realExecutableDirectory != nullptr );
+        
         EXPECT_STRCASEEQ(realExecutableDirectory, absolutePath->c_str());
+
+        if (realExecutableDirectory != nullptr)
+        {
+            free(realExecutableDirectory);
+        }
```

Comment:
This nullptr check is unneeded. The [free](https://en.cppreference.com/w/cpp/memory/c/free) function does nothing if it is supplied a nullptr.